oteapi/datacache/datacache.py

```diff
@@ -95,8 +94,8 @@ def __init__(
     ) -> None:
         if config is None:
             self.config = DataCacheConfig()
-        elif isinstance(config, dict):
-            self.config = DataCacheConfig(**config, extra=Extra.ignore)
+        elif isinstance(config, (Mapping, AttrDict)):
```

Comment:
```suggestion
        elif isinstance(config, (dict, AttrDict)):
```
Not sure `Mapping` is allowed for this (although the tests seem to pass - but if this is the case, we should just use `Mapping` here and make `AttrDict` a subclass of `Mapping` as well as `BaseModel`).